lib/hammer_cli/ssloptions.rb

```diff
@@ -1,31 +1,40 @@
 module HammerCLI
   class SSLOptions
-    DEFAULT_SSL_CA_PATH = "~/.hammer/certs"
+    DEFAULT_LOCAL_CA_STORE_PATH = "~/.hammer/certs"
 
     def initialize(settings = HammerCLI::Settings, logger = Logging.logger['SSLoptions'])
       @settings = settings
       @logger = logger
     end
 
-    def self.get_options(settings = HammerCLI::Settings, logger = Logging.logger['SSLoptions'])
-      self.new(settings, logger).get_options
+    def get_local_ca_store_path
+      @settings.get(:ssl, :local_ca_store_path) || DEFAULT_LOCAL_CA_STORE_PATH
     end
 
-    def get_options
+    def get_options(uri = nil)
       ssl_options = {}
       for sslopt in [:ssl_ca_file, :ssl_ca_path, :verify_ssl] do
         ssloptval = read_ssl_option(sslopt)
         ssl_options[sslopt] = ssloptval unless ssloptval.nil?
       end
       ssl_options.merge!(cert_key_options)
 
-      ssl_options[:ssl_ca_path] = DEFAULT_SSL_CA_PATH if ssl_options[:ssl_ca_path].nil?
+      # enable ssl verification
+      ssl_options[:verify_ssl] = true if ssl_options[:verify_ssl].nil?
+
+      if ssl_options[:verify_ssl] && uri && !ssl_options[:ssl_ca_file] && !ssl_options[:ssl_ca_path]
+        uri = URI.parse(uri) if uri.is_a?(String)
+        ca_manager = HammerCLI::CACertManager.new(DEFAULT_LOCAL_CA_STORE_PATH)
```

Comment:
Could you inject cert manager with default value of `HammerCLI::CACertManager.new(DEFAULT_LOCAL_CA_STORE_PATH)`? 
It's nicely isolated functionality so it's a pity to hardcode the instance here.